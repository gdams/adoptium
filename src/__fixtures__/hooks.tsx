import {
  Binary,
  ContributorApiResponse,
  Contributor,
  LatestTemurin,
  MarketplaceRelease,
  News,
  ReleaseAsset,
  TemurinReleases,
} from '../hooks';

export const createRandomContributorApiData = (): ContributorApiResponse => ({
  login: 'login_mock',
  id: 0,
  node_id: 'node_id_mock',
  avatar_url: 'avatar_url_mock',
  gravatar_id: 'gravatar_id_mock',
  url: 'url_mock',
  html_url: 'html_url_mock',
  followers_url: 'followers_url_mock',
  following_url: 'following_url_mock',
  gists_url: 'gists_url_mock',
  starred_url: 'starred_url_mock',
  subscriptions_url: 'subscriptions_url_mock',
  organizations_url: 'organizations_url_mock',
  repos_url: 'repos_url_mock',
  events_url: 'events_url_mock',
  received_events_url: 'received_events_url_mock',
  type: 'type_mock',
  site_admin: false,
  contributions: 3,
});

export const createRandomTemurinRelease = (installer, id): ReleaseAsset => ({
  os: 'os_mock',
  architecture: 'arch_mock',
  release_link: new URL('https://release_link_mock'),
  platform_name: `platform_name_mock${id}`,
  release_name: 'release_name_mock',
  release_date: new Date(Date.UTC(2020, 0, 1)),
  binaries: [
    {
      type: 'type_mock',
      link: new URL('https://link_mock'),
      checksum: 'checksum_mock',
      size: 0,
      extension: 'extension_mock',
      installer_link: installer ? new URL('https://installer_link_mock') : undefined,
      installer_checksum: installer ? 'installer_checksum_mock' : undefined,
      installer_size: installer ? 0 : undefined,
      installer_extension: installer ? 'installer_extension_mock' : undefined,
    },
  ],
})

export const createRandomMarketplaceRelease = (installer): MarketplaceRelease => ({
  release_name: 'release_name_mock',
  vendor: 'vendor_mock',
  binary: {
    os: 'os_mock',
    architecture: 'arch_mock',
    distribution: 'distribution_mock',
    image_type: 'type_mock',
    jvm_impl: 'jvm_impl_mock',
    timestamp: new Date(Date.UTC(2020, 0, 1)),
    scm_ref: 'scm_ref_mock',
    openjdk_scm_ref: 'openjdk_scm_ref_mock',
    aqavit_results_link: new URL('https://aqavit_results_link_mock'),
    tck_affidavit_link: new URL('https://tck_affidavit_link_mock'),
    package: {
      name: 'name_mock.tar.gz',
      link: new URL('https://link_mock'),
      sha265sum: 'sha265sum_mock',
      sha256sum_link: new URL('https://sha256sum_link_mock'),
      signature_link: new URL('https://signature_link_mock'),
    },
    installer: installer
      ? [
          {
            name: 'installer_name_mock.msi',
            link: new URL('https://installer_link_mock'),
            sha265sum: 'installer_sha265sum_mock',
            sha256sum_link: new URL('https://installer_sha256sum_link_mock'),
            signature_link: new URL('https://installer_signature_link_mock'),
          },
        ]
      : undefined,
  },
  version: {
    major: 0,
    minor: 0,
    security: 0,
    patch: 0,
    build: 0,
    openjdk_version: 'openjdk_version_mock',
  }
})

export const mockLatestTemurin = (installer): LatestTemurin => ({
  download_count: 0,
  id: 'id_mock',
  release_link: new URL('https://release_link_mock'),
  release_name: 'release_name_mock',
  release_type: 'release_type_mock',
  timestamp: new Date(Date.UTC(2020, 0, 1)),
  updated_at: new Date(Date.UTC(2020, 0, 1)),
  vendor: 'vendor_mock',
  version_data: {
    major: 0,
    minor: 0,
    security: 0,
    patch: 0,
    build: 0,
    openjdk_version: 'openjdk_version_mock',
  },
  binaries: [
    {
      os: 'os_mock',
      architecture: 'arch_mock',
      image_type: 'type_mock',
      heap_size: 'heap_size_mock',
      download_count: 0,
      jvm_impl: 'jvm_impl_mock',
      package: {
        name: 'name_mock.tar.gz',
        link: new URL('https://link_mock'),
        checksum: 'sha265sum_mock',
        checksum_link: new URL('https://sha256sum_link_mock'),
        metadata_link: new URL('https://metadata_link_mock'),
        signature_link: new URL('https://signature_link_mock'),
        size: 0,
        download_count: 0,
      },
      installer: installer
      ?
        {
          name: 'installer_name_mock.msi',
          link: new URL('https://installer_link_mock'),
          checksum: 'installer_sha265sum_mock',
          checksum_link: new URL('https://installer_sha256sum_link_mock'),
          metadata_link: new URL('https://installer_metadata_link_mock'),
          signature_link: new URL('https://installer_signature_link_mock'),
          size: 0,
          download_count: 0,
        }
      : undefined,
    }
  ]
})

export const createRandomTemurinReleases = (installer, id): TemurinReleases => ({
  release_name: `release_name_mock_${id}`,
  release_link: new URL('https://release_link_mock'),
  source_url: new URL('https://source_url_mock'),
  timestamp: new Date(Date.UTC(2020, 0, 1)),
  platforms: {
    'platform_mock': {
      assets: [
        {
          os: 'os_mock',
          architecture: 'architecture_mock',
          type: 'type_mock',
          link: new URL('https://link_mock'),
          checksum: `checksum_mock${id}`,
          size: 0,
          extension: 'extension_mock',
          installer_link: installer ? new URL('https://installer_link_mock') : undefined,
          installer_checksum: installer ? 'installer_checksum_mock' : undefined,
          installer_size: installer ? 0 : undefined,
          installer_extension: installer ? 'installer_extension_mock' : undefined,
        },
      ],
    },
  },
})

export const createRandomContributorViewData = (): Contributor => ({
  avatarUri: 'avatarUri_mock',
  profileUri: 'profileUri_mock',
  login: 'login_mock',
  contributionsCount: 0,
  commitsListUri: 'commitsListUri_mock',
  repo: 'reponame_mock',
});

export const createRandomLatestForOSData = (): Binary => ({
  release_name: 'release_name_mock',
  link: new URL('https://link_mock'),
});

export const createRandomNewsAndEventsData = (): News => ({
  news: [
    {
      id: 'id_mock',
      title: 'title_mock',
      body: 'body_mock',
      date: new Date(Date.UTC(2020, 0, 1)),
      link: 'link_mock',
    },
  ],
  events: [
    {
      id: 'id_mock',
      title: 'title_mock',
      infoLink: 'infoLink_mock',
      date: new Date(Date.UTC(2020, 0, 1)),
    },
  ]
});

export const createRandomContributorViewData1Contribution = (): Contributor => ({
  avatarUri: 'avatarUri_mock',
  profileUri: 'profileUri_mock',
  login: 'login_mock',
  contributionsCount: 1,
  commitsListUri: 'commitsListUri_mock',
  repo: 'reponame_mock',
});