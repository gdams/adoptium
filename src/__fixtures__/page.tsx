import {
    AsciidocPage,
    File,
} from '../types';

export const createAsciidocData = (): {
    asciidoc: AsciidocPage;
    file: File;
  } => ({
    asciidoc: {
        id: 'asciidoc-1',
        html: '<p>Hello World, this is a <pre><code>test</code></pre></p>',
        document: {
            title: 'Asciidoc Page title',
            main: 'Asciidoc Page content',
        },
        pageAttributes: {
            authors: 'author1, author2',
        },
        fields: {
            slug: '/asciidoc/asciidoc-page-title',
        }
    },
    file: {
        relativePath: 'test.adoc',
    }
});
