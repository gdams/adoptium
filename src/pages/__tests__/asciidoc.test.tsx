import React from 'react';
import { render } from '@testing-library/react';
import AllAsciidocPages from '../../templates/asciidocTemplate';
import { axe, toHaveNoViolations } from 'jest-axe';
import { createAsciidocData } from '../../__fixtures__/page';
import '../../../test/__mocks__/intersectionObserverMock';

expect.extend(toHaveNoViolations);
let mockData = createAsciidocData();

describe('Asciidoc pages', () => {
  it('renders correctly', () => {
    const { container } = render(<AllAsciidocPages data={mockData} />);
    // eslint-disable-next-line
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });

  it('has no accessibility violations', async () => {
    const { container } = render(<AllAsciidocPages data={mockData} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});