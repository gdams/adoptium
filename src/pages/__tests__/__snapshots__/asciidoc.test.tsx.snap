// Vitest Snapshot v1

exports[`Asciidoc pages > renders correctly 1`] = `
<main>
  <title>
    Asciidoc Page title | Adoptium
  </title>
  <meta
    content="Sample Description"
    name="description"
  />
  <meta
    content="Asciidoc Page title | Adoptium"
    name="og:title"
  />
  <meta
    content="Sample Description"
    name="og:description"
  />
  <meta
    content="website"
    name="og:type"
  />
  <meta
    content="https://adoptium.net/images/social-image.png"
    name="og:image"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="@adoptium"
    name="twitter:site"
  />
  <meta
    content="https://adoptium.net/images/social-image.png"
    name="twitter:image"
  />
  <meta
    content="@adoptium"
    name="twitter:creator"
  />
  <meta
    content="Asciidoc Page title | Adoptium"
    name="twitter:title"
  />
  <meta
    content="Sample Description"
    name="twitter:description"
  />
  <link
    href="//www.eclipse.org/eclipse.org-common/themes/solstice/public/stylesheets/vendor/cookieconsent/cookieconsent.min.css"
    rel="stylesheet"
    type="text/css"
  />
  <script
    src="//www.eclipse.org/eclipse.org-common/themes/solstice/public/javascript/vendor/cookieconsent/default.min.js"
  />
  <section
    class="py-5 container"
  >
    <div
      class="asciidoc-container container-adoc"
      id="asciidoc-container"
    >
      <div
        class="asciidoc"
      >
        <h1
          class="pb-4 fw-light text-center"
        >
          Asciidoc Page title
        </h1>
        <div
          class="asciidoc-content"
        >
          <p>
            Hello World, this is a 
          </p>
          <pre>
            <code
              class="hljs language-bash"
            >
              <span
                class="hljs-built_in"
              >
                test
              </span>
            </code>
          </pre>
          <p />
        </div>
      </div>
    </div>
    <hr
      class="m-5"
    />
    <p
      class="h5"
    >
      Contributors
    </p>
    <ul
      class="list m-0"
    >
      <li>
        <a
          class="link"
          href="https://github.com/author1"
          rel="noopener noreferrer"
          style="margin-left: 0px;"
          target="_blank"
          title="author1"
        >
          <img
            alt="author1"
            class="img"
            src="https://github.com/author1.png?size=60"
          />
        </a>
      </li>
      <li>
        <a
          class="link"
          href="https://github.com/author2"
          rel="noopener noreferrer"
          target="_blank"
          title=" author2"
        >
          <img
            alt=" author2"
            class="img"
            src="https://github.com/author2.png?size=60"
          />
        </a>
      </li>
    </ul>
    <div
      class="mx-auto pt-5"
    >
      <span
        class="fw-bold"
      >
        Help us make these docs great!
      </span>
      <br />
      <span>
        All Adoptium docs are open source. See something that's wrong or unclear?
      </span>
      <br />
      <a
        class="btn btn-secondary mt-2"
        href="https://github.com/adoptium/website-v2/edit/main/src/asciidoc-pages/test.adoc"
        rel="noopener noreferrer"
        style="border-color: #0869da; background-color: rgb(246, 248, 250);"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="18"
          stroke="currentColor"
          stroke-width="0"
          style="margin-right: .3em;"
          viewBox="0 0 512 512"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 376.6V206.8c0-46.4-3.7-70.8-22.1-94C374 87.7 348.9 77 310 77h-22V16l-96 96 96 96v-65h22c24.6 0 29 3.6 33.8 9.6 5.5 6.9 8.2 19.1 8.2 54.2v169.8c-19.1 11.1-32 31.7-32 55.4 0 35.3 28.7 64 64 64s64-28.7 64-64c0-23.7-12.9-44.3-32-55.4zM384 472c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zM128 48c-35.3 0-64 28.7-64 64 0 23.7 12.9 44.3 32 55.4v209.2c-19.1 11.1-32 31.7-32 55.4 0 35.3 28.7 64 64 64s64-28.7 64-64c0-23.7-12.9-44.3-32-55.4V167.4c19.1-11.1 32-31.7 32-55.4 0-35.3-28.7-64-64-64zm0 424c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40zm0-320c-22.1 0-40-17.9-40-40s17.9-40 40-40 40 17.9 40 40-17.9 40-40 40z"
          />
        </svg>
        Edit this page
      </a>
    </div>
  </section>
</main>
`;
